### PR TITLE
Reolink floodlighs

### PR DESCRIPTION
**Pull request by apocaliss92**
Add support to reolink lights

### CODE REVIEW

File: plugins/reolink/src/main.ts [hunk 1]
```diff
@@ -295,6 +316,14 @@ class ReolinkCamera extends RtspSmartCamera implements Camera, DeviceProvider, R
             && this.storageSettings.values.abilities?.value?.Ability?.supportAudioAlarm?.ver !== 0;
     }
 
+    hasFloodlight() {
+        const channel = this.getRtspChannel();
+        const floodLightConfigVer = this.storageSettings.values.abilities?.value?.Ability?.abilityChn?.[channel].floodLight?.ver ?? 0;
```
Comment (koush) [inline plugins/reolink/src/main.ts:321]: optional chain the index. use the same variable so you're not evaluating the same thing 3 times.

File: plugins/reolink/src/main.ts [hunk 2]
```diff
@@ -766,40 +795,71 @@ class ReolinkCamera extends RtspSmartCamera implements Camera, DeviceProvider, R
...
+            return this.floodlight;
         }
     }
 
     async releaseDevice(id: string, nativeId: string) {
         if (nativeId.endsWith('-siren')) {
             delete this.siren;
-        }
+        } else
+            if (nativeId.endsWith('--floodlight')) {
```
Comment (koush) [inline plugins/reolink/src/main.ts:860]: why 2 dashes

File: plugins/reolink/src/reolink-api.ts [hunk 3]
```diff
@@ -431,6 +431,59 @@ export class ReolinkCameraClient {
...
+            responseType: 'json',
+        }, this.createReadable(body));
+
+        const error = response.body?.[0]?.error;
+        if (error) {
+            this.console.error('error during call to setWhiteLedState', JSON.stringify(body), error);
+        }
+    }
+
+    async getWhiteLedState() {
```
Comment (koush) [inline plugins/reolink/src/reolink-api.ts:464]: this does not look to be called, and I dont see anywhere that this value is polled to update the state from outside of scrypted?

File: plugins/reolink/src/main.ts (hunk 1 above)
Comment (apocaliss92) [reply plugins/reolink/src/main.ts]: done

File: plugins/reolink/src/main.ts (hunk 2 above)
Comment (apocaliss92) [reply plugins/reolink/src/main.ts]: wops, fixed

File: plugins/reolink/src/reolink-api.ts (hunk 3 above)
Comment (apocaliss92) [reply plugins/reolink/src/reolink-api.ts]: yes was a leftover, removed

File: plugins/reolink/src/main.ts [hunk 4]
```diff
@@ -295,6 +316,21 @@ class ReolinkCamera extends RtspSmartCamera implements Camera, DeviceProvider, R
             && this.storageSettings.values.abilities?.value?.Ability?.supportAudioAlarm?.ver !== 0;
     }
 
+    hasFloodlight() {
+        const channel = this.getRtspChannel();
+
+        const channelData = this.storageSettings.values.abilities?.value?.Ability?.abilityChn?.[channel];
+        if (channelData) {
+            const floodLightConfigVer = channelData.floodLight?.ver ?? 0;
```
Comment (koush) [inline plugins/reolink/src/main.ts:324]: you need optional chaining on channelData.

Comment (apocaliss92) [reply plugins/reolink/src/main.ts:324]: It's checked in the if, isn't it?

**Pull request merged**